db/migrate/20231103165752_add_hmis_activity_log.rb

```diff
@@ -0,0 +1,19 @@
+class AddHmisActivityLog < ActiveRecord::Migration[6.1]
+  def change
+    create_table :hmis_activity_logs do |t|
+      t.references :user, null: false, index: true
+      t.references :data_source, null: false, index: false
+      t.jsonb :resolved_fields
```

Comment:
This is cool, great to see that we can store which clients were returned from a search query too.

I'm a bit worried about the size of this field though. As an example, below is the `resolved_fields` from _one of_ the logs that gets stored when I load a household's assessments.

Maybe I missed some discussion while I was out, but do we have a demonstrated need to store with **fields** were resolved on each object? I can only think of one case where _might_ want that: Client DOB and SSN (extra sensitive and have specific permissions to protect them). I can't think of a use-case for other records. To me it seems sufficient to store basically just the _keys_ in this object (`Client/1, IncomeBenefit/2, Enrollment/3, etc`. @ttoomey @eanders can you explain the thought behind that?


(As an aside, another thing I noticed is a lot of repetition in the fields. For example, `"User/775": ["id", "name", "id", "name", "id", "name"]`. Maybe this is pointing to some issue with our queries? )


<details>
  <summary>GetHouseholdAssessments resolved fields</summary>

   ```
{
  "User/768": [
    "id",
    "name"
  ],
  "User/775": [
    "id",
    "name",
    "id",
    "name",
    "id",
    "name"
  ],
  "Client/2272": [
    "id"
  ],
  "Client/2676": [
    "id"
  ],
  "Client/5698": [
    "id"
  ],
  "Client/5907": [
    "id"
  ],
  "Enrollment/59090": [
    "id",
    "lockVersion",
    "entryDate",
    "exitDate",
    "client",
    "disablingCondition",
    "dateOfEngagement",
    "moveInDate",
    "livingSituation",
    "rentalSubsidyType",
    "lengthOfStay",
    "losUnderThreshold",
    "previousStreetEssh",
    "dateToStreetEssh",
    "timesHomelessPastThreeYears",
    "monthsHomelessPastThreeYears",
    "enrollmentCoc",
    "dateOfPathStatus",
    "clientEnrolledInPath",
    "reasonNotEnrolled",
    "percentAmi",
    "referralSource",
    "countOutreachReferralApproaches",
    "dateOfBcpStatus",
    "eligibleForRhy",
    "reasonNoServices",
    "runawayYouth",
    "sexualOrientation",
    "sexualOrientationOther",
    "formerWardChildWelfare",
    "childWelfareYears",
    "childWelfareMonths",
    "formerWardJuvenileJustice",
    "juvenileJusticeYears",
    "juvenileJusticeMonths",
    "unemploymentFam",
    "mentalHealthDisorderFam",
    "physicalDisabilityFam",
    "alcoholDrugUseDisorderFam",
    "insufficientIncome",
    "incarceratedParent",
    "targetScreenReqd",
    "timeToHousingLoss",
    "annualPercentAmi",
    "literalHomelessHistory",
    "clientLeaseholder",
    "hohLeaseholder",
    "subsidyAtRisk",
    "evictionHistory",
    "criminalRecord",
    "incarceratedAdult",
    "prisonDischarge",
    "sexOffender",
    "disabledHoh",
    "currentPregnant",
    "singleParent",
    "dependentUnder6",
    "hh5Plus",
    "cocPrioritized",
    "hpScreeningScore",
    "thresholdScore",
    "vamcStation",
    "translationNeeded",
    "preferredLanguage",
    "preferredLanguageDifferent",
    "customDataElements"
  ],
  "Enrollment/59092": [
    "id",
    "lockVersion",
    "entryDate",
    "exitDate",
    "disablingCondition",
    "dateOfEngagement",
    "moveInDate",
    "livingSituation",
    "rentalSubsidyType",
    "lengthOfStay",
    "losUnderThreshold",
    "previousStreetEssh",
    "dateToStreetEssh",
    "timesHomelessPastThreeYears",
    "monthsHomelessPastThreeYears",
    "enrollmentCoc",
    "dateOfPathStatus",
    "clientEnrolledInPath",
    "reasonNotEnrolled",
    "percentAmi",
    "referralSource",
    "countOutreachReferralApproaches",
    "dateOfBcpStatus",
    "eligibleForRhy",
    "reasonNoServices",
    "runawayYouth",
    "sexualOrientation",
    "sexualOrientationOther",
    "formerWardChildWelfare",
    "childWelfareYears",
    "childWelfareMonths",
    "formerWardJuvenileJustice",
    "juvenileJusticeYears",
    "juvenileJusticeMonths",
    "unemploymentFam",
    "mentalHealthDisorderFam",
    "physicalDisabilityFam",
    "alcoholDrugUseDisorderFam",
    "insufficientIncome",
    "incarceratedParent",
    "targetScreenReqd",
    "timeToHousingLoss",
    "annualPercentAmi",
    "literalHomelessHistory",
    "clientLeaseholder",
    "hohLeaseholder",
    "subsidyAtRisk",
    "evictionHistory",
    "criminalRecord",
    "incarceratedAdult",
    "prisonDischarge",
    "sexOffender",
    "disabledHoh",
    "currentPregnant",
    "singleParent",
    "dependentUnder6",
    "hh5Plus",
    "cocPrioritized",
    "hpScreeningScore",
    "thresholdScore",
    "vamcStation",
    "translationNeeded",
    "preferredLanguage",
    "preferredLanguageDifferent",
    "customDataElements",
    "client"
  ],
  "Enrollment/59356": [
    "id",
    "lockVersion",
    "entryDate",
    "exitDate",
    "disablingCondition",
    "dateOfEngagement",
    "moveInDate",
    "livingSituation",
    "rentalSubsidyType",
    "lengthOfStay",
    "losUnderThreshold",
    "previousStreetEssh",
    "dateToStreetEssh",
    "timesHomelessPastThreeYears",
    "monthsHomelessPastThreeYears",
    "enrollmentCoc",
    "dateOfPathStatus",
    "clientEnrolledInPath",
    "reasonNotEnrolled",
    "percentAmi",
    "referralSource",
    "countOutreachReferralApproaches",
    "dateOfBcpStatus",
    "eligibleForRhy",
    "reasonNoServices",
    "runawayYouth",
    "sexualOrientation",
    "sexualOrientationOther",
    "formerWardChildWelfare",
    "childWelfareYears",
    "childWelfareMonths",
    "formerWardJuvenileJustice",
    "juvenileJusticeYears",
    "juvenileJusticeMonths",
    "unemploymentFam",
    "mentalHealthDisorderFam",
    "physicalDisabilityFam",
    "alcoholDrugUseDisorderFam",
    "insufficientIncome",
    "incarceratedParent",
    "targetScreenReqd",
    "timeToHousingLoss",
    "annualPercentAmi",
    "literalHomelessHistory",
    "clientLeaseholder",
    "hohLeaseholder",
    "subsidyAtRisk",
    "evictionHistory",
    "criminalRecord",
    "incarceratedAdult",
    "prisonDischarge",
    "sexOffender",
    "disabledHoh",
    "currentPregnant",
    "singleParent",
    "dependentUnder6",
    "hh5Plus",
    "cocPrioritized",
    "hpScreeningScore",
    "thresholdScore",
    "vamcStation",
    "translationNeeded",
    "preferredLanguage",
    "preferredLanguageDifferent",
    "customDataElements",
    "client"
  ],
  "Enrollment/59357": [
    "id",
    "lockVersion",
    "entryDate",
    "exitDate",
    "disablingCondition",
    "dateOfEngagement",
    "moveInDate",
    "livingSituation",
    "rentalSubsidyType",
    "lengthOfStay",
    "losUnderThreshold",
    "previousStreetEssh",
    "dateToStreetEssh",
    "timesHomelessPastThreeYears",
    "monthsHomelessPastThreeYears",
    "enrollmentCoc",
    "dateOfPathStatus",
    "clientEnrolledInPath",
    "reasonNotEnrolled",
    "percentAmi",
    "referralSource",
    "countOutreachReferralApproaches",
    "dateOfBcpStatus",
    "eligibleForRhy",
    "reasonNoServices",
    "runawayYouth",
    "sexualOrientation",
    "sexualOrientationOther",
    "formerWardChildWelfare",
    "childWelfareYears",
    "childWelfareMonths",
    "formerWardJuvenileJustice",
    "juvenileJusticeYears",
    "juvenileJusticeMonths",
    "unemploymentFam",
    "mentalHealthDisorderFam",
    "physicalDisabilityFam",
    "alcoholDrugUseDisorderFam",
    "insufficientIncome",
    "incarceratedParent",
    "targetScreenReqd",
    "timeToHousingLoss",
    "annualPercentAmi",
    "literalHomelessHistory",
    "clientLeaseholder",
    "hohLeaseholder",
    "subsidyAtRisk",
    "evictionHistory",
    "criminalRecord",
    "incarceratedAdult",
    "prisonDischarge",
    "sexOffender",
    "disabledHoh",
    "currentPregnant",
    "singleParent",
    "dependentUnder6",
    "hh5Plus",
    "cocPrioritized",
    "hpScreeningScore",
    "thresholdScore",
    "vamcStation",
    "translationNeeded",
    "preferredLanguage",
    "preferredLanguageDifferent",
    "customDataElements",
    "client"
  ],
  "Assessment/238841": [
    "id",
    "lockVersion",
    "inProgress",
    "assessmentDate",
    "dataCollectionStage",
    "dateCreated",
    "dateUpdated",
    "dateDeleted",
    "role",
    "access",
    "wipValues",
    "youthEducationStatus",
    "employmentEducation",
    "customDataElements",
    "user",
    "enrollment",
    "incomeBenefit",
    "healthAndDv",
    "exit",
    "disabilityGroup"
  ],
  "Assessment/238843": [
    "id",
    "lockVersion",
    "inProgress",
    "assessmentDate",
    "dataCollectionStage",
    "dateCreated",
    "dateUpdated",
    "dateDeleted",
    "role",
    "access",
    "wipValues",
    "youthEducationStatus",
    "employmentEducation",
    "customDataElements",
    "user",
    "enrollment",
    "incomeBenefit",
    "healthAndDv",
    "exit",
    "disabilityGroup"
  ],
  "Assessment/238901": [
    "id",
    "lockVersion",
    "inProgress",
    "assessmentDate",
    "dataCollectionStage",
    "dateCreated",
    "dateUpdated",
    "dateDeleted",
    "role",
    "access",
    "wipValues",
    "youthEducationStatus",
    "employmentEducation",
    "customDataElements",
    "user",
    "enrollment",
    "incomeBenefit",
    "healthAndDv",
    "exit",
    "disabilityGroup"
  ],
  "Assessment/238902": [
    "id",
    "lockVersion",
    "inProgress",
    "assessmentDate",
    "dataCollectionStage",
    "dateCreated",
    "dateUpdated",
    "dateDeleted",
    "role",
    "access",
    "wipValues",
    "youthEducationStatus",
    "employmentEducation",
    "customDataElements",
    "user",
    "enrollment",
    "incomeBenefit",
    "healthAndDv",
    "exit",
    "disabilityGroup"
  ],
  "HealthAndDv/77102": [
    "currentlyFleeing",
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dentalHealthStatus",
    "domesticViolenceSurvivor",
    "dueDate",
    "generalHealthStatus",
    "id",
    "informationDate",
    "mentalHealthStatus",
    "pregnancyStatus",
    "whenOccurred"
  ],
  "HealthAndDv/87506": [
    "currentlyFleeing",
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dentalHealthStatus",
    "domesticViolenceSurvivor",
    "dueDate",
    "generalHealthStatus",
    "id",
    "informationDate",
    "mentalHealthStatus",
    "pregnancyStatus",
    "whenOccurred"
  ],
  "HealthAndDv/87507": [
    "currentlyFleeing",
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dentalHealthStatus",
    "domesticViolenceSurvivor",
    "dueDate",
    "generalHealthStatus",
    "id",
    "informationDate",
    "mentalHealthStatus",
    "pregnancyStatus",
    "whenOccurred"
  ],
  "IncomeBenefit/81752": [
    "adap",
    "alimony",
    "alimonyAmount",
    "benefitsFromAnySource",
    "childSupport",
    "childSupportAmount",
    "cobra",
    "connectionWithSoar",
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "earned",
    "earnedAmount",
    "employerProvided",
    "ga",
    "gaAmount",
    "id",
    "incomeFromAnySource",
    "indianHealthServices",
    "informationDate",
    "insuranceFromAnySource",
    "medicaid",
    "medicare",
    "noAdapReason",
    "noCobraReason",
    "noEmployerProvidedReason",
    "noIndianHealthServicesReason",
    "noMedicaidReason",
    "noMedicareReason",
    "noPrivatePayReason",
    "noRyanWhiteReason",
    "noSchipReason",
    "noStateHealthInsReason",
    "noVhaReason",
    "otherBenefitsSource",
    "otherBenefitsSourceIdentify",
    "otherIncomeAmount",
    "otherIncomeSource",
    "otherIncomeSourceIdentify",
    "otherInsurance",
    "otherInsuranceIdentify",
    "otherTanf",
    "pension",
    "pensionAmount",
    "privateDisability",
    "privateDisabilityAmount",
    "privatePay",
    "ryanWhiteMedDent",
    "schip",
    "snap",
    "socSecRetirement",
    "socSecRetirementAmount",
    "ssdi",
    "ssdiAmount",
    "ssi",
    "ssiAmount",
    "stateHealthIns",
    "tanf",
    "tanfAmount",
    "tanfChildCare",
    "tanfTransportation",
    "totalMonthlyIncome",
    "unemployment",
    "unemploymentAmount",
    "vaDisabilityNonService",
    "vaDisabilityNonServiceAmount",
    "vaDisabilityService",
    "vaDisabilityServiceAmount",
    "vhaServices",
    "wic",
    "workersComp",
    "workersCompAmount",
    "customDataElements"
  ],
  "IncomeBenefit/81754": [
    "adap",
    "alimony",
    "alimonyAmount",
    "benefitsFromAnySource",
    "childSupport",
    "childSupportAmount",
    "cobra",
    "connectionWithSoar",
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "earned",
    "earnedAmount",
    "employerProvided",
    "ga",
    "gaAmount",
    "id",
    "incomeFromAnySource",
    "indianHealthServices",
    "informationDate",
    "insuranceFromAnySource",
    "medicaid",
    "medicare",
    "noAdapReason",
    "noCobraReason",
    "noEmployerProvidedReason",
    "noIndianHealthServicesReason",
    "noMedicaidReason",
    "noMedicareReason",
    "noPrivatePayReason",
    "noRyanWhiteReason",
    "noSchipReason",
    "noStateHealthInsReason",
    "noVhaReason",
    "otherBenefitsSource",
    "otherBenefitsSourceIdentify",
    "otherIncomeAmount",
    "otherIncomeSource",
    "otherIncomeSourceIdentify",
    "otherInsurance",
    "otherInsuranceIdentify",
    "otherTanf",
    "pension",
    "pensionAmount",
    "privateDisability",
    "privateDisabilityAmount",
    "privatePay",
    "ryanWhiteMedDent",
    "schip",
    "snap",
    "socSecRetirement",
    "socSecRetirementAmount",
    "ssdi",
    "ssdiAmount",
    "ssi",
    "ssiAmount",
    "stateHealthIns",
    "tanf",
    "tanfAmount",
    "tanfChildCare",
    "tanfTransportation",
    "totalMonthlyIncome",
    "unemployment",
    "unemploymentAmount",
    "vaDisabilityNonService",
    "vaDisabilityNonServiceAmount",
    "vaDisabilityService",
    "vaDisabilityServiceAmount",
    "vhaServices",
    "wic",
    "workersComp",
    "workersCompAmount",
    "customDataElements"
  ],
  "IncomeBenefit/92185": [
    "adap",
    "alimony",
    "alimonyAmount",
    "benefitsFromAnySource",
    "childSupport",
    "childSupportAmount",
    "cobra",
    "connectionWithSoar",
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "earned",
    "earnedAmount",
    "employerProvided",
    "ga",
    "gaAmount",
    "id",
    "incomeFromAnySource",
    "indianHealthServices",
    "informationDate",
    "insuranceFromAnySource",
    "medicaid",
    "medicare",
    "noAdapReason",
    "noCobraReason",
    "noEmployerProvidedReason",
    "noIndianHealthServicesReason",
    "noMedicaidReason",
    "noMedicareReason",
    "noPrivatePayReason",
    "noRyanWhiteReason",
    "noSchipReason",
    "noStateHealthInsReason",
    "noVhaReason",
    "otherBenefitsSource",
    "otherBenefitsSourceIdentify",
    "otherIncomeAmount",
    "otherIncomeSource",
    "otherIncomeSourceIdentify",
    "otherInsurance",
    "otherInsuranceIdentify",
    "otherTanf",
    "pension",
    "pensionAmount",
    "privateDisability",
    "privateDisabilityAmount",
    "privatePay",
    "ryanWhiteMedDent",
    "schip",
    "snap",
    "socSecRetirement",
    "socSecRetirementAmount",
    "ssdi",
    "ssdiAmount",
    "ssi",
    "ssiAmount",
    "stateHealthIns",
    "tanf",
    "tanfAmount",
    "tanfChildCare",
    "tanfTransportation",
    "totalMonthlyIncome",
    "unemployment",
    "unemploymentAmount",
    "vaDisabilityNonService",
    "vaDisabilityNonServiceAmount",
    "vaDisabilityService",
    "vaDisabilityServiceAmount",
    "vhaServices",
    "wic",
    "workersComp",
    "workersCompAmount",
    "customDataElements"
  ],
  "IncomeBenefit/92186": [
    "adap",
    "alimony",
    "alimonyAmount",
    "benefitsFromAnySource",
    "childSupport",
    "childSupportAmount",
    "cobra",
    "connectionWithSoar",
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "earned",
    "earnedAmount",
    "employerProvided",
    "ga",
    "gaAmount",
    "id",
    "incomeFromAnySource",
    "indianHealthServices",
    "informationDate",
    "insuranceFromAnySource",
    "medicaid",
    "medicare",
    "noAdapReason",
    "noCobraReason",
    "noEmployerProvidedReason",
    "noIndianHealthServicesReason",
    "noMedicaidReason",
    "noMedicareReason",
    "noPrivatePayReason",
    "noRyanWhiteReason",
    "noSchipReason",
    "noStateHealthInsReason",
    "noVhaReason",
    "otherBenefitsSource",
    "otherBenefitsSourceIdentify",
    "otherIncomeAmount",
    "otherIncomeSource",
    "otherIncomeSourceIdentify",
    "otherInsurance",
    "otherInsuranceIdentify",
    "otherTanf",
    "pension",
    "pensionAmount",
    "privateDisability",
    "privateDisabilityAmount",
    "privatePay",
    "ryanWhiteMedDent",
    "schip",
    "snap",
    "socSecRetirement",
    "socSecRetirementAmount",
    "ssdi",
    "ssdiAmount",
    "ssi",
    "ssiAmount",
    "stateHealthIns",
    "tanf",
    "tanfAmount",
    "tanfChildCare",
    "tanfTransportation",
    "totalMonthlyIncome",
    "unemployment",
    "unemploymentAmount",
    "vaDisabilityNonService",
    "vaDisabilityNonServiceAmount",
    "vaDisabilityService",
    "vaDisabilityServiceAmount",
    "vhaServices",
    "wic",
    "workersComp",
    "workersCompAmount",
    "customDataElements"
  ],
  "CustomDataElement/13": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/14": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/15": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/16": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/17": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/18": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/19": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/20": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/25": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "CustomDataElement/27": [
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "key",
    "label",
    "fieldType",
    "repeats",
    "values",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value",
    "id",
    "value"
  ],
  "EmploymentEducation/81648": [
    "dataCollectionStage",
    "dateCreated",
    "dateDeleted",
    "employed",
    "employmentType",
    "id",
    "informationDate",
    "lastGradeCompleted",
    "notEmployedReason",
    "schoolStatus"
  ]
}
   ```
</details>

<details>
  <summary>client search resolved fields</summary>
```
{
  "Client/1216": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/2272": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/6477": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/8017": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/13731": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/15550": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/17066": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/17752": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/19271": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "Client/27768": [
    "id",
    "lockVersion",
    "firstName",
    "middleName",
    "lastName",
    "nameSuffix",
    "age",
    "access",
    "dateCreated",
    "dateDeleted",
    "dateUpdated",
    "dob",
    "ssn",
    "externalIds"
  ],
  "ExternalIdentifier/client_id:1216": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:2272": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:6477": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:8017": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:13731": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:15550": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:17066": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:17752": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:19271": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/client_id:27768": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:31514": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:32497": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:37035": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:38235": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:41770": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:43502": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:44230": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:44300": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:46528": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/warehouse_id:54999": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:04ad6a64a29a1ea3822cf6376d938b70": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:0855601064a8baf509c3b87361bfc4ec": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:48784ec2e9fba5e378bf90e6ed215502": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:6cd9313ed34ef58bad3fdd504355e72c": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:77a09250b8aa75054a97b0a3747dff59": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:8a7f3325cfeac84ddd117ac84ef0c124": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:933e5dafaccb0d1e7c243daa081c9359": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:a328b64fb251b3891b369070736a2b21": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:d1f2b9b49b2883e1c6098072d4ae95c0": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ],
  "ExternalIdentifier/personal_id:e057698ae77cc136e1ae4aa0a173dfca": [
    "id",
    "identifier",
    "url",
    "label",
    "type"
  ]
}
```
</details>

Reply:
> Maybe I missed some discussion while I was out, but do we have a demonstrated need to store with fields were resolved on each object? 

@gigxz it was an assumption on my part that it was desirable to have the capability for granular field-level reporting. My thinking is that it's better to have logs and not need them than the other way around.

However if we are confident that field-level reporting will never be a requirement, I'm comfortable reducing the fields logged. I suggest we keep the current structure to allow tracking restricted fields or other fields in the future:
```json
{
  "Client/1": ["SSN"],
  "Project/1": []
}
```

> As an aside, another thing I noticed is a lot of repetition in the fields

Good spot, thanks! Yeah it might be we are resolving the same object through different paths in a query. Regardless I should dedup those before storing

Reply:
Okay, gotcha. What do you think @eanders? I like the idea of keeping this structure, but only including certain fields that are deemed audit-worthy.

Reply:
Well I think my suggestion would work for that. If DisablingCondition (or some similar field) was resolved, that indicates that the "full" enrollment record is being viewed.

Reply:
okay, added `activity_log_field_name` to filter out most fields. The only fields I'm logging are on client as follows:
'ssn', 'dob', 'disabilities', 'disability_groups'